src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -286,24 +291,26 @@ public async Task EmissionsBatchActual_MissingRequiredParams_ReturnsBadRequest(b
        
         var intesityData = new List<CarbonIntensityBatchDTO>() { intensityBatch };
 
-        var result = await PostJSONBodyToURI(intesityData, batchActualURI);
+        var result = await PostJSONBodyToURI(intesityData, batchAverageCarbonIntensityURI);
 
         Assert.That(result, Is.Not.Null);
         Assert.That(result?.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
     }
 
-    [TestCase("2022-1-1T04:05:06Z", "2022-1-2T04:05:06Z", "eastus", 1, TestName = "EmissionsBatchActual expects OK for eastus 1 element")]
-    [TestCase("2021-12-25", "2021-12-26", "westus", 3, TestName = "EmissionsBatchActual expects OK for westus 3 elements")]
-    public async Task EmissionsBatchActual_SupportedDataSources_ReturnsOk(DateTimeOffset start, DateTimeOffset end, string location, int nelems)
+    [TestCase("2022-1-1T04:05:06Z", "2022-1-2T04:05:06Z", "eastus", 1, TestName = "EmissionsMarginalCarbonIntensityBatch expects OK for eastus 1 element")]
+    [TestCase("2021-12-25", "2021-12-26", "westus", 3, TestName = "EmissionsMarginalCarbonIntensityBatch expects OK for westus 3 elements")]
+    public async Task EmissionsMarginalCarbonIntensityBatch_SupportedDataSources_ReturnsOk(string start, string end, string location, int nelems)
     {
-        _dataSourceMocker.SetupDataMock(start, end, location);
+        var startDate = DateTimeOffset.Parse(start);
+        var endDate = DateTimeOffset.Parse(end);
+        _dataSourceMocker.SetupDataMock(startDate, endDate, location);
         var intesityData = Enumerable.Range(0, nelems).Select(x => new CarbonIntensityBatchDTO() 
```

Comment:
```suggestion
        var intesityData = Enumerable.Range(0, nelems).Select(x => new {
            location = location,
            startTime = start,
            endTime = end
        }); 
```

plus cleanup below because I can't do multi-line suggestions